### PR TITLE
Fix/prepare upgrade

### PR DESCRIPTION
**Pull request by tituomin**
Fix all deprecation warnings except one in munigeo to prepare in upgrading Django.

### CODE REVIEW

File: observations/models.py [hunk 1]
```diff
@@ -76,21 +77,25 @@ class Observation(PolymorphicModel):
     """
     value = models.ForeignKey(
         AllowedValue, blank=False, null=True,
+        on_delete=models.CASCADE,
```
Comment (Rikuoja) [inline observations/models.py]: Here, it seems that corresponding observations are deleted if an allowed value is deleted. Perhaps we want to make sure that no observations use a value before deleting it?

File: observations/models.py [hunk 2]
```diff
@@ -76,21 +77,25 @@ class Observation(PolymorphicModel):
...
         related_name='instances')
     time = models.DateTimeField(
         db_index=True,
         help_text='Exact time the observation was made')
     unit = models.ForeignKey(
         services_models.Unit, blank=False, null=False,
+        on_delete=models.CASCADE,
```
Comment (Rikuoja) [inline observations/models.py]: Do we want to retain observation history even if units are deleted? Then we should probably soft delete units instead?

File: observations/models.py [hunk 3]
```diff
@@ -76,21 +77,25 @@ class Observation(PolymorphicModel):
...
         help_text='The unit the observation is about',
         related_name='observation_history')
     units = models.ManyToManyField(
         services_models.Unit, through='UnitLatestObservation')
     auth = models.ForeignKey(
-        'PluralityAuthToken', null=True)
+        'PluralityAuthToken', null=True,
+        on_delete=models.CASCADE)
```
Comment (Rikuoja) [inline observations/models.py]: Do we want to retain observation history even if a user (or a "plurality") token is deleted? Can we implement soft deletes for plurality tokens, or do plurality tokens have the same rights under GDPR to be forgotten?

File: observations/models.py [hunk 4]
```diff
@@ -76,21 +77,25 @@ class Observation(PolymorphicModel):
...
     property = models.ForeignKey(
         ObservableProperty,
         blank=False, null=False,
+        on_delete=models.CASCADE,
```
Comment (Rikuoja) [inline observations/models.py]: Should we prevent deleting observable properties if somebody has made observations already? Soft delete perhaps?

File: services/management/commands/verify_school_districts.py [hunk 5]
```diff
@@ -96,5 +73,18 @@ def test__verify_school_units_enclosed(division_units):
                     'start': division.start, 'unit': unit, 'geom': unit.location.wkt,
                     'unit.srid': unit.location.srid, 'div.srid': division.geometry.boundary.srid})
             success = False
-    assert success, (
+    return success, (
         "{} errors \n".format(error_count) + "\n\n".join([pprint.pformat(error, indent=4) for error in error_report]))
+
+
+class Command(BaseCommand):
+    help = """ Verify that imported school district data correctly refers to
```
Comment (Rikuoja) [inline services/management/commands/verify_school_districts.py]: Perhaps instead create a dummy smbackend importer that 1) calls the munigeo division importer first, and 2) then verifies the imported data?

File: observations/models.py (hunk 1 above)
Comment (tituomin) [reply observations/models.py]: Observation deletion prevented in 35306b441fac9c4352c57a3873c114a2e1912663
Actual soft deletion added to backlog.

File: observations/models.py (hunk 2 above)
Comment (tituomin) [reply observations/models.py]: Observation deletion prevented in 35306b441fac9c4352c57a3873c114a2e1912663
Actual soft deletion added to backlog.

File: observations/models.py (hunk 3 above)
Comment (tituomin) [reply observations/models.py]: Observation deletion prevented in 35306b441fac9c4352c57a3873c114a2e1912663
Actual soft deletion added to backlog.

File: observations/models.py (hunk 4 above)
Comment (tituomin) [reply observations/models.py]: Observation deletion prevented in 35306b441fac9c4352c57a3873c114a2e1912663
Actual soft deletion added to backlog.

File: services/management/commands/verify_school_districts.py (hunk 5 above)
Comment (tituomin) [reply services/management/commands/verify_school_districts.py]: Added to backlog. The data verification wasn't run automatically before this PR, so making it run is another task anyway.

**Pull request merged**